db/commitlog/commitlog.cc

```diff
@@ -1143,7 +1143,9 @@ db::commitlog::segment_manager::segment_manager(config c)
     // our threshold for trying to force a flush. needs heristics, for now max - segment_size/2.
     , disk_usage_threshold(cfg.commitlog_flush_threshold_in_mb.has_value() 
         ? size_t(std::ceil(*cfg.commitlog_flush_threshold_in_mb / double(smp::count))) * 1024 * 1024 
-        : (max_disk_size - (max_disk_size > (max_size/2) ? (max_size/2) : 0)))
+        : (max_disk_size -
+            (max_disk_size >= (max_size*2) ? max_size
+                : (max_disk_size > (max_size/2) ? (max_size/2) : max_disk_size/3))))
```

Comment:
I have to say I'm dazed by the calculation.

I also think it should be a percentage of the commitlog size, not a segment. There's no reason to expect correlation between memtable flush time and a segment size. We should probably flush at around 75% full. I don't mind doing that later though.

Reply:
Agree. This is just trying to make it something like "we are in the last segment".

Reply:
The threshold calculation doesn't make sense to me.

If `max_disk_size >= (max_size*2)` the threshold will be at 50% or higher of max_disk_size.
But if `(max_size/2) < max_disk_size < (max_size*2)` the threshold could be very close to 100% if max_disk_size is only slightly larger than `max_size/2`.  Did you mean to compare to `max_size` instead? Like this:

```
        : (max_disk_size -
            (max_disk_size >= (max_size*2) ? max_size
                : (max_disk_size > max_size ? (max_size/2) : max_disk_size/3))))
```